### ISSUE TITLE
best_hyperparameters output confusing -- a bug?

### ISSUE THREAD

**Issue by amjass12** [keras-team/keras-tuner#378]
Hi all, 

I have a question regarding the `tuner.get_best_hyperparameters `output that is slightly confusing. I am using this output to carry out an independent training on the same parameters to do a 10 fold cross_val. After calling tuner.search as follows:

```
tuner.search(X_train.values, y_train,
             epochs=200,
             validation_split=0.1,
             callbacks=[EarlyStopping(monitor='val_loss', patience=5)], shuffle=True)

```
I save the best model but also the corresponding best_parameters as follows:
```

tuner.results_summary()
best_model = tuner.get_best_models(1)[0]

best_hyperparameters = tuner.get_best_hyperparameters(1)[0]
bestConfig=best_hyperparameters.get_config()

```
I then save the bestConfig as a text file.. this works fine, however, the optimal model contains extra information (layers?).. the summary for the best_model is:
```

Model: "sequential"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
dense (Dense)                (None, 832)               17278976  
_________________________________________________________________
dense_1 (Dense)              (None, 976)               813008    
_________________________________________________________________
dense_2 (Dense)              (None, 880)               859760    
_________________________________________________________________
dense_3 (Dense)              (None, 944)               831664    
_________________________________________________________________
dense_4 (Dense)              (None, 19)                17955     
=================================================================
Total params: 19,801,363
Trainable params: 19,801,363
Non-trainable params: 0
```
The output (part of) of the best parameters is as follows:

values,{**'num_layers': 4, 'units_0': 832, 'units_1': 976, 'learning_rate': 0.0001, 'units_2': 880, 'units_3': 944, 'units_4': 864**, 'units_5': 352, 'units_6': 592, 'units_7': 432, 'units_8': 112, 'units_9': 320, 'units_10': 752, 'units_11': 496, 'units_12': 352, 'units_13': 528, 'units_14': 352, 'units_15': 800, 'units_16': 896, 'units_17': 656, 'units_18': 960}

As you can see the first units (0-3) are correct according to the best_model summary, and I am guessing that the learning rate is correct (the learning rate used in the model was 0.0001?)

What I am unsure of is, what are the units after units_3? It is as if there were an additional set of layers in the model?

I am sorry if i have missed something very obvious!

**Comment by yixingfu** [comment]
It will display all hyperparameters added along the way of tuning, no matter whether it is used or not, as long as it is considered `active`. 

The tuner is unable to automatically infer whether a hyperparameter is truely active or not, unless you tell it by explicitly using conditional scope (see [this example 8 and 9](https://github.com/keras-team/keras-tuner/blob/master/examples/helloworld.py#L213), you may need to use github head instead of released version).  However, no matter you go with this approach or not,  the extra hyperparameters will not affect the tuning outcome.

**Comment by amjass12** [comment]
Hi @yixingfu ,

thank you very much for clarifying! this makes sense! so just to confirm, this would mean the final model summary are the 'best' parameters found for this model and the ones after are the hyperparamaters that were added during tuning?

thanks!!

**Comment by yixingfu** [comment]
Right. For example in first trial it tries to search for 8 layers, which add hyperparameters like `units_7`. Then, say, the best result has 4 layers in total, then for that best trial when it builds the model it only uses up to `units_3` but the hp object still keeps track of `units_4` to `units_7`, even if they do nothing to the specific model.

**Comment by amjass12** [comment]
got it!! thank you very much for your time and for clarifying! :)
